src/collections/Menu/MenuItem.js

```diff
@@ -8,16 +8,21 @@ import {
 } from '../../lib'
 
 function MenuItem(props) {
-  const { active, children, className, header } = props
+  const { active, children, className, header, href, link, onClick } = props
   const classes = cx(
     className,
     useKeyOnly(active, 'active'),
     useKeyOnly(header, 'header'),
+    useKeyOnly(link, 'link'),
     'item',
   )
+  const handleClick = (e) => {
+    if (onClick) onClick(e)
+  }
   const rest = getUnhandledProps(MenuItem, props)
+  const ItemComponent = href || onClick ? 'a' : 'div'
 
-  return <a {...rest} className={classes}>{children}</a>
+  return <ItemComponent {...rest} className={classes} onClick={handleClick}>{children}</ItemComponent>
```

Comment:
As we pass `onClick` [here](https://github.com/TechnologyAdvice/stardust/pull/382/commits/f39a7463dd65e3590311519afe15724eb9ce8a9c#diff-151cbde1a0eb8a3f890a84401e62f45cR88), `ItemComponent` will be defined always as `a`. Ideas?